### ISSUE TITLE
`bash -c` hang forever

### ISSUE THREAD

**Issue by purocean** [Microsoft/WSL#2460]
* Your Windows build number:  Microsoft Windows [10.0.16278.1000]

* What you're doing and what's happening:  
Today I upgrade Windows 10 to 10.0.16278.1000, my *VSCode* some commands not work.

* Test node script:
```js
const spawn = require('child_process').spawn;

const dir = spawn('cmd', ['/c', 'dir']);
dir.stdout.on('data', (data) => {
  console.log(`>>>>>stdout: ${data}`);
});

dir.stderr.on('data', (data) => {
  console.log(`>>>>>>stderr: ${data}`);
});

dir.on('close', () => {
  console.log(`>>>>>>>close`);
  console.log('------------------------------------------------');
});

dir.on('exit', () => {
  console.log(`>>>>>>>exit`);
});

dir.on('disconnect', () => {
  console.log(`>>>>>>>disconnect`);
});

//////////////////////////////////////////////////////////

const ls = spawn('bash', ['-c', 'ls']);

ls.stdout.on('data', (data) => {
  console.log(`>>>>>stdout: ${data}`);
});

ls.stderr.on('data', (data) => {
  console.log(`>>>>>>stderr: ${data}`);
});

ls.on('close', () => {
  console.log(`>>>>>>>close`);
});

ls.on('exit', () => {
  console.log(`>>>>>>>exit`);
});

ls.on('disconnect', () => {
  console.log(`>>>>>>>disconnect`);
});
```

![image](https://user-images.githubusercontent.com/7115690/29870890-3fd6723a-8dbc-11e7-92e1-5bc1fb79de8b.png)

**Nerver exit child process!**

**Comment by purocean** [comment]
![image](https://user-images.githubusercontent.com/7115690/29871951-39793d56-8dc0-11e7-91bb-5cc682170c26.png)


I use *cmd* be a proxy for bash，it works
```
const dir = spawn('cmd', ['/c', 'bash -c ls']);
```
![image](https://user-images.githubusercontent.com/7115690/29871979-4dd3c622-8dc0-11e7-9d76-b4bd93875c38.png)

**Comment by alexstepankov** [comment]
Same situation, ide(phpstorm, vscode) show error when call bash -c (or `ubuntu run` and etc) only on build 16278

https://youtrack.jetbrains.com/issue/WI-37962

**Comment by benhillis** [comment]
@purocean - This issue has been fixed internally and will be available in a Windows Insider build sometime soon.

**Comment by riartem** [comment]
I have this issue too.. For example, when I added File Watcher for compiling "less" files in PHPStorm, it calls and executed but shown as infinity loading
![2017-10-27_16-51-48](https://user-images.githubusercontent.com/987517/32107426-32ed6620-bb37-11e7-98e6-f40c57919e01.gif)

**Comment by hurricup** [comment]
This is https://github.com/Microsoft/WSL/issues/2592

**Comment by riartem** [comment]
@benhillis I think it might be closed only after including fix to main updates channel.

**Comment by benhillis** [comment]
Closed because it is a duplicate of another issue.
